python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -248,6 +248,7 @@ def add_output(self, ref: RefBundle) -> None:
         """Move a bundle produced by the operator to its outqueue."""
         self.outqueue.append(ref)
         self.num_completed_tasks += 1
+        self.op.metrics.rows_task_outputs_generated += ref.num_rows() or 0
```

Comment:
Let's instead 

 - Move this into the base implementation of `PhysicalOperator.get_next`
 - Abstract a method in metrics, rather than modifying counter internally